src/transformers/modeling_tf_longformer.py

```diff
@@ -0,0 +1,1377 @@
+                all_attentions = all_attentions + (layer_outputs[1],)
+
+        # Add last layer
+        if cast_bool_to_primitive(output_hidden_states, self.output_hidden_states) is True:
+            hidden_states_to_add = hidden_states[:, :-padding_len] if padding_len > 0 else hidden_states
+            all_hidden_states = all_hidden_states + (hidden_states_to_add,)
+
+        outputs = (hidden_states,)
+        if cast_bool_to_primitive(output_hidden_states, self.output_hidden_states) is True:
+            outputs = outputs + (all_hidden_states,)
+        if cast_bool_to_primitive(output_attentions, self.output_attentions) is True:
+            outputs = outputs + (all_attentions,)
+        return outputs  # outputs, (hidden states), (attentions)
+
+
+@keras_serializable
+class TFLongformerMainLayer(tf.keras.layers.Layer):
+    config_class = LongformerConfig
+
+    def __init__(self, config, **kwargs):
+        super().__init__(**kwargs)
+
+        if isinstance(config.attention_window, int):
+            assert config.attention_window % 2 == 0, "`config.attention_window` has to be an even value"
+            assert config.attention_window > 0, "`config.attention_window` has to be positive"
+            config.attention_window = [config.attention_window] * config.num_hidden_layers  # one value per layer
+        else:
+            assert len(config.attention_window) == config.num_hidden_layers, (
+                "`len(config.attention_window)` should equal `config.num_hidden_layers`. "
+                f"Expected {config.num_hidden_layers}, given {len(config.attention_window)}"
+            )
+
+        self.num_hidden_layers = config.num_hidden_layers
+        self.initializer_range = config.initializer_range
+        self.output_attentions = config.output_attentions
+        self.output_hidden_states = config.output_hidden_states
+        self.pad_token_id = config.pad_token_id
+        self.attention_window = config.attention_window
+
+        self.embeddings = TFRobertaEmbeddings(config, name="embeddings")
+        self.encoder = TFLongformerEncoder(config, name="encoder")
+        self.pooler = TFBertPooler(config, name="pooler")
+
+    def get_input_embeddings(self):
+        return self.embeddings
+
+    def set_input_embeddings(self, value):
+        self.embeddings.word_embeddings = value
+        self.embeddings.vocab_size = value.shape[0]
+
+    def _prune_heads(self, heads_to_prune):
+        """ Prunes heads of the model.
+            heads_to_prune: dict of {layer_num: list of heads to prune in this layer}
+            See base class PreTrainedModel
+        """
+        raise NotImplementedError
+
+    def call(
+        self,
+        inputs,
+        attention_mask=None,
+        global_attention_mask=None,
+        token_type_ids=None,
+        position_ids=None,
+        inputs_embeds=None,
+        output_attentions=None,
+        output_hidden_states=None,
+        training=False,
+    ):
```

Comment:
Would be great to handle the newly merged `return_dict` here too

Reply:
+1

Reply:
I'll add it :-)